component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/MixinCleanerUpgradePlugin.java

```diff
@@ -0,0 +1,362 @@
+import org.exoplatform.services.jcr.impl.core.NodeImpl;
+import org.exoplatform.services.jcr.impl.core.SessionImpl;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.transaction.TransactionService;
+
+/**
+ * Created by The eXo Platform SAS Author : Boubaker Khanfir
+ * bkhanfir@exoplatform.com April 16, 2016
+ */
+public class MixinCleanerUpgradePlugin extends UpgradeProductPlugin {
+
+  public static final String        DEFAULT_WORKSPACE_NAME         = "social";
+
+  public static final String        MIGRATION_STATUS               = "migration.status";
+
+  public static final int           UPDATE_LAST_NODE_FREQ          = 1000;
+
+  private static final int          TRANSACTION_TIMEOUT_IN_SECONDS = 86400;
+
+  private static final Log          LOG                            = ExoLogger.getLogger(MixinCleanerUpgradePlugin.class);
+
+  private static final int          NODES_IN_ONE_TRANSACTION       = 100;
+
+  private final PortalContainer     portalContainer;
+
+  private final RepositoryService   repositoryService;
+
+  private final TransactionService  txService;
+
+  private String                    workspaceName;
+
+  private long                      totalCount                     = 0;
+
+  private Map<String, List<String>> mixinNames                     = null;
+
+  private String                    jcrRootPath;
+
+  private long                      maxTreatedNodes                = 0;
+
+  private boolean                   upgradeFinished                = false;
+
+  /**
+   * @param portalContainer
+   * @param repositoryService
+   * @param txService
+   * @param initParams workspace: workspace on which the operation will start.
+   *          mixins.to.clean, mixins.clean.exception
+   */
+  public MixinCleanerUpgradePlugin(PortalContainer portalContainer,
+                                   RepositoryService repositoryService,
+                                   TransactionService txService,
+                                   SettingService settingService,
+                                   ChromatticManager chromatticManager,
+                                   InitParams initParams) {
+    super(settingService, chromatticManager, initParams);
+    this.repositoryService = repositoryService;
+    this.txService = txService;
+    this.portalContainer = portalContainer;
+    ValueParam workspaceValueParam = initParams.getValueParam("workspace");
+    if (workspaceValueParam != null) {
+      workspaceName = workspaceValueParam.getValue();
+    }
+    if (StringUtils.isBlank(workspaceName)) {
+      workspaceName = DEFAULT_WORKSPACE_NAME;
+    }
+    ValueParam pathParam = initParams.getValueParam("path");
+    if (pathParam != null) {
+      jcrRootPath = pathParam.getValue();
+    }
+    if (StringUtils.isBlank(jcrRootPath)) {
+      jcrRootPath = "/";
+    }
+
+    ValuesParam mixinsValueParam = initParams.getValuesParam("mixins.to.clean");
```

Comment:
We should have consistent and more meaningful param names IMO. What about :
* mixinsCleanup.includes
* mixinsCleanup.maxNodes
* mixinsCleanup.excludes
?